src/olympia/devhub/tasks.py

```diff
@@ -742,3 +744,20 @@ def send_api_key_revocation_email(emails):
         use_deny_list=False,
         perm_setting='individual_contact',
     )
+
+
+@task
+@use_primary_db
+def create_site_permission_version(
```

Comment:
Why are we using a task for this, as opposed to just doing it synchronously when the form is submitted? Is it because we have to do signing asynchronously?

Reply:
signing would take place in `auto_approve` asynchronously no matter what - it will never be immediate.

The reason why I switched to a task was because behind the scenes I'm reusing the regular primitives we se for uploads, the zip is created the zip in memory, written to the temporary location for file uploads, then to the final location, and there are a few database calls in between to set up everything. All trivial operations, but it can add up under load and I don't want the http servers to wait for that.

All in all it's consistent with everything else in addons-server: things that can take time/resources in a significant way are in a task or cronjob.